src/main/java/com/minecolonies/commands/ColonyInfo.java

```diff
@@ -0,0 +1,146 @@
+    private static final String NO_COLONY_FOUND_MESSAGE_ID = "Colony with ID %d not found.";
+
+    public static final String DESC                        = "info";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public ColonyInfo(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        int colonyId = -1;
+        UUID mayorID = sender.getCommandSenderEntity().getUniqueID();
+        final boolean found;
+
+        if (args.length != 0)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                if(!args[0].isEmpty())
+                {
+                    mayorID = getUUIDFromString(sender, args);
+                }
+            }
+        }
+
+        final IColony tempColony;
+        if(colonyId == -1)
+        {
+            tempColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID);
+        }
+        else
+        {
+            tempColony = ColonyManager.getColony(colonyId);
+        }
+
+        if(tempColony == null)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(NO_COLONY_FOUND_MESSAGE_ID, colonyId)));
+            return;
+        }
+
+        Colony colony = ColonyManager.getColony(sender.getEntityWorld(), tempColony.getCenter());
+        if (colony == null)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(NO_COLONY_FOUND_MESSAGE_ID, colonyId)));
+            return;
+        }
+
+        colonyId = colony.getID();
+        if (colonyId == -1)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(NO_COLONY_FOUND_MESSAGE, args[0])));
+            return;
+        }
+        
+        final BlockPos position = colony.getCenter();
+        sender.addChatMessage(new TextComponentString(ID_TEXT + colony.getID() + NAME_TEXT + colony.getName()));
+        final String mayor = colony.getPermissions().getOwnerName();
+        sender.addChatMessage(new TextComponentString(MAYOR_TEXT + mayor));
+        sender.addChatMessage(new TextComponentString(CITIZENS + colony.getCitizens().size() + "/" + colony.getMaxCitizens()));
+        sender.addChatMessage(new TextComponentString(COORDINATES_TEXT + String.format(COORDINATES_XYZ, position.getX(), position.getY(), position.getZ())));
+    }
+
+    private static UUID getUUIDFromString(@NotNull final ICommandSender sender, @NotNull final String... args)
```

Comment:
getUUIDFromName, this method confused be at first because I didn't know why we weren't just using UUID.fromString. The method name implies that that we are converting a String(UUID as a string) to a UUID, but we are getting the UUID of a player from their name.